egs/babel_uam/asr1/run.sh

```diff
@@ -0,0 +1,406 @@
+      local/phone_maps/${l} 
+    
+    cat ${dict}/{,non}silence_lexicon.txt | sort > ${dict}/lexicon.txt
+    
+    # Prepare the rest of the dictionary directory
+    # -----------------------------------------------
+    # The local/prepare_dict.py script, which is basically the same as
+    # prepare_unicode_lexicon.py used in the babel recipe to create the
+    # graphemic lexicons, is better suited for working with kaldi formatted
+    # lexicons and can be used for this task by only modifying optional input
+    # arguments. If we could modify local/prepare_lexicon.pl to accomodate this
+    # need it may be more intuitive.
+    ./local/prepare_dict.py \
+      --silence-lexicon ${dict}/silence_lexicon.txt ${dict}/lexicon.txt ${dict}
+
+    ###########################################################################
+    # Prepend language ID to all utterances to disambiguate between speakers
+    # of different languages sharing the same speaker id.
+    #
+    # The individual lang directories can be used for alignments, while a
+    # combined directory will be used for training. This probably has minimal
+    # impact on performance as only words repeated across languages will pose
+    # problems and even amongst these, the main concern is the <hes> marker.
+    ###########################################################################
+    echo "Prepend ${l} to data dir"
+    ./utils/copy_data_dir.sh --spk-prefix "${l}_" --utt-prefix "${l}_" \
+      data/train data/train_${l}
+      cd $cwd
+  fi
+  cd ${cwd}
+done
+
+###############################################################################
+# Combine all langauge specific training directories and generate a single
+# lang directory by combining all langauge specific dictionaries
+###############################################################################
+if [ $stage -le 2 ]; then
+  train_dirs=""
+  dict_dirs=""
+  for l in ${langs}; do
+    train_dirs="data/${l}/data/train_${l} ${train_dirs}"
+    dict_dirs="data/${l}/data/dict_universal ${dict_dirs}"
+  done
+
+  ./utils/combine_data.sh data/train $train_dirs
+
+  # This script was made to mimic the utils/combine_data.sh script, but instead
+  # it merges the lexicons while reconciling the nonsilence_phones.txt,
+  # silence_phones.txt, and extra_questions.txt by basically just calling
+  # local/prepare_unicode_lexicon.py. As mentioned, it may be better to simply
+  # modify an existing script to automatically create the dictionary dir from
+  # a lexicon, rather than overuse the local/prepare_unicode_lexicon.py script.
+  ./local/combine_lexicons.sh data/dict_universal $dict_dirs
+
+  # Prepare lang directory
+  ./utils/prepare_lang.sh --share-silence-phones true \
+    data/dict_universal "<unk>" data/dict_universal/tmp.lang data/lang_universal
+fi
+
+
+###############################################################################
+# TRAINING FOR PHONE ALIGNMENTS
```

Comment:
I was planning on it actually. That was one of the first things I had planned for today.

Reply:
Also, I don't think it depends on the CLSP grid structure. If you're referring to the many configurations files. Most of those just contain the many paths to the many datasets corresponding to each language. I agree that it is really annoying to have to change all of those, but I can't think of a good alternative either. I can probably remove some of the files and paths that point to unused things though. Do you have any ideas?

Reply:
OK. The issue is that compared with other `run.sh` in ESPnet, the setting of corpus directory is not included, and it might confuse people. (I actually still don't exactly find where I set the corpus directory, as I only checking this repo, and did not check the actual files, pointed by symlinks of this recipe. Is `babel/s5d/conf/lang/*.conf` the one that we specify the directory?). I like your design of making use of existing Kaldi recipe files with symlinks, but this corpus directory setup requires special knowledge of the Kaldi babel recipe. Maybe the solution is to add some comments in `run.sh` about how people set the corpus directory and so on for this recipe.

Reply:
Yes. babel/s5d/conf/lang/*.conf contain all of the language specific configurations including paths to directories that need to be set. The problem is that for each language I need to set the directories for the 10h training set, 10h dev set and lexicon file. 

And since there are 25 languages, that would mean setting 75 paths. What about if I just create one large file with all the paths to all of the files and source that?

Reply:
I like your idea.
Put this corpus setup file in `egs/babel_uam/asr1/conf` (or `egs/babel_uam/asr1/local`) with some comments about this corpus path setup in `run.sh`. 
I think this is a good solution.